lib/handler/cache-handler.js

```diff
@@ -0,0 +1,355 @@
+'use strict'
+
+const util = require('../core/util')
+const DecoratorHandler = require('../handler/decorator-handler')
+const { parseCacheControlHeader, parseVaryHeader } = require('../util/cache')
+
+/**
+ * Writes a response to a CacheStore and then passes it on to the next handler
+ */
+class CacheHandler extends DecoratorHandler {
+  /**
+   * @type {import('../../types/cache-interceptor.d.ts').default.CacheOptions}
+   */
+  #opts = null
```

Comment:
As I suggested, why should #opts be stored and not the values?

Reply:
Honestly personal preference and makes it easier to pass around but not needed